src/components/HorizontalStepper/HorizontalStepper.tsx

```diff
@@ -0,0 +1,82 @@
+import clsx from 'clsx';
+import React from 'react';
+import styles from './HorizontalStepper.module.css';
+import HorizontalStep from '../HorizontalStep';
+
+export interface Props {
+  /**
+   * Zero-based index that identifies which index is the active step.
+   */
+  activeIndex: number;
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * Ordered list of step texts.
+   */
+  steps: string[];
+}
+
+/**
+ * ```ts
+ * import {HorizontalStepper} from "@chanzuckerberg/eds";
+ * ```
+ *
+ * A stepper component used to display which steps have been completed, the current active step, and possible remaining steps.
+ *
+ * Example usage:
+ *
+ * ```tsx
+ * <HorizontalStepper activeIndex={0} steps={['Step 1', 'Step 2', 'Step 3']} />
+ * ```
+ */
+export const HorizontalStepper = ({ activeIndex, className, steps }: Props) => {
+  if (
+    process.env.NODE_ENV !== 'production' &&
+    !(activeIndex >= 0 && activeIndex <= steps.length)
```

Comment:
Conditional written as such to also account for `NaN` values which pass the Typescript type check for `number`